### PR TITLE
feat(subscriptions): enable annual subscriptions

### PR DESCRIPTION
**Pull request by olarclara**
## What kind of change does this PR introduce?

Closes [XTD-696](https://linear.app/codesandbox/issue/XTD-696/release-0-allow-admin-to-open-stripe-to-check-out-on-annual).

Updates the workspace settings UI to allow free teams to go through the checkout process and purchase seats on yearly subscriptions.

https://user-images.githubusercontent.com/24959348/221274141-a0301700-5320-4fff-9c10-e1e9a3bf8fa0.mov

Updates the new team creation to include more subscription options:

localhost_3000_dashboard_settings_workspace=f9da145f-8e41-4e57-85d8-a361fd6def83

Updates the `/pro` page to show subscription options:

Screen Shot 2023-02-24 at 20 33 30

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

1. Launch the dashboard
2. Visit a team that does not have an active subscription
3. Go to the settings
4. Click on the caret in the upgrade card
5. Select "Custom upgrade"
6. The UI in Stripe should allow to select the quantity of seats to purchase

## Checklist

- [ ] Documentation
- [ ] Testing 
- [ ] Ready to be merged
      
- [ ] Added myself to contributors table

### CODE REVIEW

**Comment by tristandubbeld** [comment]
Should we include the extra options on the CTA to start a trial for teams that are still eligible for one? Found this one on the settings page:

image

File: packages/app/src/app/pages/Dashboard/Components/TeamSubscriptionOptions/TeamSubscriptionOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  buttonVariant,
+  buttonStyles,
+  ctaCopy,
+  trackingLocation,
+}) => {
+  const { activeTeam } = useAppState();
+  const { isAdmin } = useWorkspaceAuthorization();
+  const { isFree, isEligibleForTrial } = useWorkspaceSubscription();
+
+  const canSubscribe = isAdmin && isFree && activeTeam !== null;
```
Comment (tristandubbeld) [inline packages/app/src/app/pages/Dashboard/Components/TeamSubscriptionOptions/TeamSubscriptionOptions.tsx:26]: If we need to ensure it's also a team and not personal, we can use `isTeamAdmin` instead.

**Review by tristandubbeld — APPROVED** [review]
Looks good to me! It seems that the buttons on the settings page do not have focus states yet, that's the only thing that caught my eye. Other than that, lets ship it!

File: packages/app/src/app/pages/Dashboard/Components/TeamSubscriptionOptions/TeamSubscriptionOptions.tsx (hunk 1 above)
Comment (olarclara) [reply packages/app/src/app/pages/Dashboard/Components/TeamSubscriptionOptions/TeamSubscriptionOptions.tsx]: @tristandubbeld thank you! I think it's better, updated here: [159eb55](https://github.com/codesandbox/codesandbox-client/pull/7455/commits/159eb5577f930192a714acac7eb2f7b4f39f2aa0).

**Comment by olarclara** [comment]
@tristandubbeld that's a good catch, I'm not sure why it doesn't show the focus styles. Created a ticket to fix it here: https://linear.app/codesandbox/issue/XTD-736/add-focus-styles-to-combobutton.

**Pull request merged**